### PR TITLE
CLN: convert str.format() to f-string

### PR DESCRIPTION
**Pull request by AlpAribal**
- [x] ref #29547
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -10424,8 +10407,8 @@ def last_valid_index(self):
 
 def _doc_parms(cls):
     """Return a tuple of the doc parms."""
-    axis_descr = "{%s}" % ", ".join(
-        "{0} ({1})".format(a, i) for i, a in enumerate(cls._AXIS_ORDERS)
+    axis_descr = (
```
Comment (jreback) [inline pandas/core/generic.py]: does this work? a nested f-string

File: pandas/core/nanops.py [hunk 2]
```diff
@@ -60,8 +60,11 @@ def __call__(self, f):
         def _f(*args, **kwargs):
             obj_iter = itertools.chain(args, kwargs.values())
             if any(self.check(obj) for obj in obj_iter):
-                msg = "reduction operation {name!r} not allowed for this dtype"
-                raise TypeError(msg.format(name=f.__name__.replace("nan", "")))
+                msg = (
+                    f"reduction operation {f.__name__.replace('nan', '')!r}"
```
Comment (jreback) [inline pandas/core/nanops.py]: use repr rather than ``!r``

File: pandas/core/nanops.py [hunk 3]
```diff
@@ -1299,9 +1302,7 @@ def _ensure_numeric(x):
                 x = complex(x)
             except ValueError:
                 # e.g. "foo"
-                raise TypeError(
-                    "Could not convert {value!s} to numeric".format(value=x)
-                )
+                raise TypeError(f"Could not convert {x!s} to numeric")
```
Comment (jreback) [inline pandas/core/nanops.py]: no need for ``!s``, you an use ``repr(x)`` if you want

File: pandas/core/resample.py [hunk 4]
```diff
@@ -963,7 +961,7 @@ def _maybe_process_deprecations(r, how=None, fill_method=None, limit=None):
 
         # .resample(..., how='sum')
         if isinstance(how, str):
-            method = "{0}()".format(how)
+            method = f"{how}()"
```
Comment (jreback) [inline pandas/core/resample.py]: leave this function entirely it is being removed

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/nanops.py (hunk 2 above)
Comment (AlpAribal) [reply pandas/core/nanops.py]: Sure, I saw another issue regarding !r, thought I shouldn't replace it with this PR.

File: pandas/core/generic.py (hunk 1 above)
Comment (AlpAribal) [reply pandas/core/generic.py]: It does work, I checked it.

File: pandas/core/nanops.py (hunk 2 above)
Comment (jreback) [reply pandas/core/nanops.py]: neh its fine to do here

File: pandas/core/nanops.py (hunk 3 above)
Comment (jreback) [reply pandas/core/nanops.py]: same, ok to change here

Comment (AlpAribal) [reply pandas/core/nanops.py]: I changed `x!s` to `repr(x)` but now a test fails. The reason is that the test expects `"foo"` whereas `repr` returns `"'foo'"`. Should I replace `repr` with `str` or get rid of `repr`?

Comment (jreback) [reply pandas/core/nanops.py]: use repr where it makes sense
if i have to fix a test that is ok

Comment (AlpAribal) [reply pandas/core/nanops.py]: The only failing test is related to this. Shall I update the test also?

**Pull request closed without merging**